src/honeysql/types.cljc

```diff
@@ -0,0 +1,84 @@
+(ns honeysql.types
```

Comment:
I just commented on the PR, but repeating here for clarity: if you run `git diff 6300ead4f994e1a4d382fb4269fac913a2874b69..d36d52ee9e0710661c693db43736207bc9e473a7` you should see only the 'real' changes, excluding the big file rename from `.clj` to `.cljc`.

This might be a PR you prefer to review commit-by-commit.